ci-operator/step-registry/ipi/conf/azure/confidential/confidentialvm/ipi-conf-azure-confidential-confidentialvm-commands.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+PATCH="/tmp/install-config-security.yaml.patch"
+
+# OCPBUGS-36670: encryptionAtHost and settings.securityType are not related.
+# feature setting of image definition should only depend on settings.securityType
+if [[ "${ENABLE_CONFIDENTIALVM_DEFAULT_MACHINE}" == "true" ]]; then
+    cat >> "${PATCH}" << EOF
+platform:
+  azure:
+    defaultMachinePlatform:
+      encryptionAtHost: false
+      settings:
+        securityType: ConfidentialVM
+        confidentialVM:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+      osDisk:
+        securityProfile:
+          securityEncryptionType: ${OS_DISK_SECURITY_ENCRYPTION_TYPE}
+EOF
+fi
+
+if [[ "${ENABLE_CONFIDENTIALVM_CONTROL_PLANE}" == "true" ]]; then
+    cat >> "${PATCH}" << EOF
+controlPlane:
+  platform:
+    azure:
+      encryptionAtHost: false
+      settings:
+        securityType: ConfidentialVM
+        confidentialVM:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+      osDisk:
+        securityProfile:
+          securityEncryptionType: ${OS_DISK_SECURITY_ENCRYPTION_TYPE}
+EOF
+fi
+
+if [[ "${ENABLE_CONFIDENTIALVM_COMPUTE}" == "true" ]]; then
+    cat >> "${PATCH}" << EOF
+compute:
+- platform:
+    azure:
+      encryptionAtHost: true
+      settings:
+        securityType: ConfidentialVM
+        confidentialVM:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+      osDisk:
+        securityProfile:
+          securityEncryptionType: ${OS_DISK_SECURITY_ENCRYPTION_TYPE}
+EOF
+fi
+
+# apply patch to install-config
+yq-go m -x -i "${CONFIG}" "${PATCH}"
+
+cat ${PATCH}
```

Comment:
I am thinking if we can merge this step with `ipi-conf-azure-confidential-trustedlaunch` as `ipi-conf-azure-confidential`.